### PR TITLE
Added support for CREATE TABLE attribute WITH JOURNAL TABLE

### PR DESCRIPTION
**Pull request by michaelvdow**

### CODE REVIEW

File: babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateAttributeJournalTable.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+public class SqlCreateAttributeJournalTable extends SqlCreateAttribute {
+
+  private final SqlIdentifier tableName;
+
+  /**
+   * Creates a {@code SqlCreateAttributeJournalTable}.
+   *
+   * @param tableName  Name of the permanent journal table to be used
+   * @param pos  Parser position, must not be null
```
Comment (AndrewPochapsky) [inline babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateAttributeJournalTable.java:35]: nit: punctuation

Comment (AndrewPochapsky) [reply babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateAttributeJournalTable.java:35]: Actually never mind, I don't think we actually add periods to @param statements

Comment (michaelvdow) [reply babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateAttributeJournalTable.java:35]: I think @param lines typically don't have a period at the end

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 2]
```diff
@@ -327,6 +327,26 @@ private void checkParseInfixCast(String sqlType) {
...
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeJournalTableWithSource() {
+    final String sql = "create table foo, with journal table = baz.tbl (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH JOURNAL TABLE = `BAZ`.`TBL` (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeMultipleAttrs() {
```
Comment (AndrewPochapsky) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:342]: Should we also test that the order of attributes doesn't matter?

Comment (michaelvdow) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:342]: That's a good point, I can add another test for that too

File: babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateAttributeJournalTable.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+   */
+  public SqlCreateAttributeJournalTable(SqlIdentifier tableName, SqlParserPos pos) {
+    super(pos);
+    this.tableName = tableName;
+  }
+
+  public void unparse(final SqlWriter writer, final int leftPrec, final int rightPrec) {
```
Comment (dasch-1) [inline babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateAttributeJournalTable.java:42]: "@Override"

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 4]
```diff
@@ -327,6 +327,34 @@ private void checkParseInfixCast(String sqlType) {
     sql(sql).ok(expected);
   }
 
+  @Test public void testTableAttributeJournalTable() {
+    final String sql = "create table foo, with journal table = baz (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH JOURNAL TABLE = `BAZ` (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeJournalTableWithSource() {
```
Comment (dasch-1) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:336]: Maybe the name of this test could be?

testTableAttributeJournalTableWithCompoundIdentifier()

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 5]
```diff
@@ -327,6 +327,34 @@ private void checkParseInfixCast(String sqlType) {
     sql(sql).ok(expected);
   }
 
+  @Test public void testTableAttributeJournalTable() {
```
Comment (dasch-1) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:330]: Maybe the name of this test could be?

testTableAttributeJournalTableWithSimpleIdentifier()

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java (hunk 2 above)
Comment (dasch-1) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:342]: Good catch, thanks!

**Pull request merged**